### ISSUE TITLE
boolean indexing

### ISSUE THREAD

**Issue by peteron30** [pydata/xarray#1486]
I am trying to figure out how boolean indexing works in xarray.
I have a couple of data arrays below:
```
X_day[latitude,longitude,time]
X_night[latitude,longitude,time]
Rule[latitude,longitude]
```

I want to merge X_day and X_night as a new X based on Rule. 
First I make a copy of X_day to be X:
`X = X_day 
`
Then I tried:
`X[Rule==True, :] = X_night 
`
and this:
`X.values[Rule==True, :] = X_night.values
`
and also this:
`X.where(Rule==True) = X_night
`
None of the above assignment worked. Please help.

**Comment by MaximilianR** [comment]
I think there are some answers on SO. 

Though no perfect solution atm, unfortunately - need to finish the work on `where`

**Comment by jhamman** [comment]
closed via #1496

@peteron30 - The next release of xarray will have a [3-argument where](http://xarray.pydata.org/en/latest/generated/xarray.where.html#xarray.where) that should serve your purposes. 

```Python
X = xr.where(Rule, X_day, X_night)
``` 

Feel free to reopen this if your use case is not met.
